addons/point_of_sale/static/src/app/models/related_models.js

```diff
@@ -474,652 +463,666 @@ export function createRelatedModels(modelDefs, modelClasses = {}, opts = {}) {
+                    const oldVal = map.get(inverse.name);
+                    map.set(inverse.name, [...(oldVal || []), record.id]);
                 }
-            } else {
-                delete crud.indexedRecords[model][key][keyVal];
-            }
-        }
+            };
 
-        return id;
-    }
+            for (const name in ownFields) {
+                const field = ownFields[name];
+                const inverse = inverseMap.get(field);
 
-    function createCRUD(model, fields) {
-        return {
-            // We need to read these object from `this` to keep
-            // the reactivity. Otherwise, we read the fields on
-            // reactive with no callback.
-            get records() {
-                return records;
-            },
-            get orderedRecords() {
-                return Array.from(this.records[model].values());
-            },
-            get indexedRecords() {
-                return indexedRecords;
-            },
-            get indexes() {
-                return indexes;
-            },
-            get modelName() {
-                return model;
-            },
-            get modelFields() {
-                return getFields(this.modelName);
-            },
-            create(vals, ignoreRelations = false, fromSerialized = false, delayedSetup = false) {
-                const record = create(
-                    this,
-                    model,
-                    vals,
-                    ignoreRelations,
-                    fromSerialized,
-                    delayedSetup
-                );
-                makeRecordsAvailable({ [model]: [record] }, { [model]: [vals] });
-                return record;
-            },
-            deserialize(vals) {
-                return deserialize(this, model, vals);
-            },
-            createMany(valsList) {
-                const result = [];
-                for (const vals of valsList) {
-                    result.push(create(this, model, vals));
-                }
-                return result;
-            },
-            update(record, vals) {
-                return update(this, model, record, vals);
-            },
-            delete(record, opts = {}) {
-                return delete_(this, model, record, opts);
-            },
-            deleteMany(toDelete) {
-                const result = [];
-                let mustBreak = 0;
-                while (toDelete.length) {
-                    result.push(delete_(this, model, toDelete[toDelete.length - 1]));
-                    mustBreak += 1;
-
-                    if (mustBreak > 1000) {
-                        console.warn("Too many records to delete. Breaking the loop.");
-                        break;
+                if (X2MANY_TYPES.has(field.type)) {
+                    for (const record2 of [...record[name]]) {
+                        handleCommand(inverse, field, record, opts.backend);
```

Comment:
i don't totally remember but is _delete recursive for sub records ?

Reply:
We're not recursively deleting. We're just unlinking the linked record from the record that is being deleted.